### PR TITLE
perf(linter): Use `aho-corasick` instead of `regex` for default comment patterns in `no-fallthrough`

### PR DESCRIPTION
**Pull request by camchenry**
theory: using Regex is slow in most cases, since we have a fixed number of patterns that we need to search for. but we do still need to support it since the user can pass a custom regex pattern. however, we can at least try to use a simpler `aho-corasick` matcher for the default strings, which should likely be the majority of cases.

### CODE REVIEW

**Comment by camchenry** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5901?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#5901** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5901?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#5892** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5892?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @camchenry and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_linter/src/rules/eslint/no_fallthrough.rs [hunk 1]
```diff
@@ -53,9 +56,18 @@ impl NoFallthrough {
...
+            None
+        };
+
         Self(Box::new(Config {
-            comment_pattern: Regex::new(format!("(?iu){comment_pattern}").as_str()).unwrap(),
+            custom_comment_pattern,
+            default_comment_pattern: AhoCorasick::builder()
+                .ascii_case_insensitive(true)
+                .build(DEFAULT_FALLTHROUGH_COMMENT_PATTERNS)
+                .expect("Could not build AhoCorasick for default comment patterns"),
```
Comment (DonIsaac) [inline crates/oxc_linter/src/rules/eslint/no_fallthrough.rs:70]: Can we provide the name of the rule so users know where and how to fix their configs?

**Comment by camchenry** [comment]
I tested this locally and it seems like this might not actually be much of a perf win. Seems like it is 2-3% slower on the large benchmarks like checker.ts and cal.com, but faster in small files (~6-7%), but not at a statistically significant level.

**Comment by DonIsaac** [comment]
This PR could still be useful if it lets us remove our dependency on `regex`

**Comment by camchenry** [comment]
@DonIsaac I think we still have to depend on regex still unfortunately, because the custom comment pattern config is allowed to be any pattern.

**Pull request closed without merging**